src/tools/CertHelperTests/KeyVaultCertTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Security.KeyVault.Certificates;
+using Azure.Security.KeyVault.Secrets;
+using Moq;
+using Xunit;
+using Xunit.Sdk;
+
+namespace CertHelper.Tests;
+
+public class KeyVaultCertTests
+{
+    [Fact]
+    public async Task GetKeyVaultCerts_ShouldAddCertificatesToCollection()
+    {
+        // Arrange
+        Mock<TokenCredential> mockTokenCred;
+        Mock<CertificateClient> mockCertClient;
+        Mock<SecretClient> mockSecretClient;
+        Mock<ILocalCert> mockLocalCert;
+        CertStoreSetup(out mockTokenCred, out mockCertClient, out mockSecretClient, out mockLocalCert, false);
+
+        var keyVaultCert = new KeyVaultCert(mockTokenCred.Object, mockCertClient.Object, mockSecretClient.Object, mockLocalCert.Object);
+
+        // Act
+        await keyVaultCert.GetKeyVaultCerts();
+
+        // Assert
+        Assert.Equal(2, keyVaultCert.KeyVaultCertificates.Count);
+    }
+
+    private static void CertStoreSetup(out Mock<TokenCredential> mockTokenCred, out Mock<CertificateClient> mockCertClient, out Mock<SecretClient> mockSecretClient, out Mock<ILocalCert> mockLocalCert, bool missingKeyVaultCerts = false, bool localAndKeyVaultDifferent = false)
+    {
+        mockTokenCred = new Mock<TokenCredential>();
+        mockTokenCred.Setup(tc => tc.GetTokenAsync(It.IsAny<TokenRequestContext>(), default)).ReturnsAsync(new AccessToken("token", DateTimeOffset.Now));
+
+        mockCertClient = new Mock<CertificateClient>(new Uri("https://dotnetperfkeyvault.vault.azure.net/"), mockTokenCred.Object);
+        mockSecretClient = new Mock<SecretClient>(new Uri("https://dotnetperfkeyvault.vault.azure.net/"), mockTokenCred.Object);
+        KeyVaultCertificateWithPolicy? mockCert1, mockCert2;
+        X509Certificate2 cert1, cert2;
+        MakeCerts(out mockCert1, out mockCert2, out cert1, out cert2, localAndKeyVaultDifferent);
+
+        var certCollection = new X509Certificate2Collection { cert1, cert2 };
+
+#pragma warning disable CS8625 // Cannot convert null literal to non-nullable reference type.
```

Comment:
I think if we are going to have multiple disables in a very short section, it would be fine to have a single one with a comment that it is for all GetCertificateAsync or whatever calls.